backend/internal/application/handler/applicationhandler.go

```diff
@@ -248,6 +264,13 @@ func (ah *ApplicationHandler) HandleApplicationDeleteRequest(w http.ResponseWrit
 	// Delete the application using the application service.
 	appProvider := appprovider.NewApplicationProvider()
 	appService := appProvider.GetApplicationService()
+
+	// Retrieve the application to log the org ID before deletion.
+	application, getErr := appService.GetApplication(id)
```

Comment:
shouldn't derive the org id from the application.. org id should be derived from the authenticated user or some other mechanism, independent of the application id..

So, lets initialize the org id, as its done in other methods..

Also lets move the initializing org id to one of the very first steps that each handle methods do.. Probably just after the initializing the logger.